rfc/src/rfcs/0005-should-fail-attr.md

```diff
@@ -0,0 +1,197 @@
+We consider that the mechanism we just mentioned and the one discussed in this proposal solve different problems, so they don't need to be discussed together.
+This is further discussed in the [rationale and alternatives](#rationale-and-alternatives) and [future possibilities](#future-possibilities) sections.
+
+## User Experience
+
+The scope of this functionality is limited to the overall verification result.
+The [rationale section](#rationale-and-alternatives) discusses the granularity of failures, and how this attribute could be extended.
+
+### Single Harness
+
+Let's take an example from the Kani regression:
+
+```rust
+#[kani::proof]
+fn simple_add_overflows() {
+    let a: u32 = kani::any();
+    let b: u32 = kani::any();
+    a + b;
+}
+```
+
+Currently, this example produces a `VERIFICATION:- FAILED` result.[^footnote-compiletest]
+In addition, it will return a non-successful code.
+
+```rust
+#[kani::proof]
+#[kani::should_fail]
+fn simple_add_overflows() {
+    let a: u32 = kani::any();
+    let b: u32 = kani::any();
+    a + b;
+}
+```
+
+Since we added `#[kani::should_fail]`, running this example would produce a successful code.
+
+Now, we've considered two ways to represent this result in the verification output.
+Note that it's important that we provide the user with this feedback:
+ 1. **(Expectation)** Kani was expecting the harness to fail.
+ 2. **(Outcome)**: The actual verification result that Kani produced after the analysis.
+This will avoid a potential scenario where the user doesn't know for sure if the attribute has had an effect when running the verification harness.
+
+As mentioned, we've considered two ways to represent this result.
+
+#### Representation #1 (Recommended): No changes to overall result
+
+```rust
+VERIFICATION:- FAILED (expected FAILED)
+```
+
+We recommend this representation so the user receives clear information about both the outcome and the expectation.
+
+#### Representation #2: Changes to overall result
+
+```rust
+VERIFICATION:- SUCCESSFUL (expected FAILED and it FAILED)
+```
+
+### Multiple Harnesses
+
+When there are multiple harnesses, we'll implement the single-harness changes in addition to the following ones.
+Currently, a "Summary" section appears after reporting the results for each harness:
+```
+Verification failed for - harness3
+Verification failed for - harness2
+Verification failed for - harness1
+Complete - 0 successfully verified harnesses, 3 failures, 3 total.
+```
+
+Harnesses marked with `#[kani::should_fail]` won't show unless the expected result was different from the actual result.
+The summary will consider harnesses that match their expectation as "successfully verified harnesses".
+
+Therefore, if we added `#[kani::should_fail]` to all harnesses in the previous example, we'd see this output:
+
+```
+Complete - 3 successfully verified harnesses, 0 failures, 3 total.
```

Comment:
Doesn't this conflict with the detailed representation?

Reply:
No, this shows up for both the default and terse outputs.